Library/Homebrew/cmd/developer.rb

```diff
@@ -0,0 +1,53 @@
+# typed: true
+# frozen_string_literal: true
+
+require "cli/parser"
+
+module Homebrew
+  extend T::Sig
+
+  module_function
+
+  sig { returns(CLI::Parser) }
+  def developer_args
+    Homebrew::CLI::Parser.new do
+      description <<~EOS
+        Control Homebrew's developer mode. When developer mode is enabled,
+        `brew update` will update Homebrew to the latest commit on the `master`
+        branch instead of the latest stable version.
```

Comment:
```suggestion
        `brew update` will update Homebrew to the latest commit on the `master`
        branch instead of the latest stable version along with some other behaviour changes.
```

Not sure if we should spell out exactly what else. I know there's stuff in `update.sh` like e.g. auto-stashing, restoring branches, etc.

Reply:
Honestly, I don't even really know what it is, so I'll just add the "other" line.